api/chains/chains.go

```diff
@@ -503,55 +503,73 @@ func VerifyTrace(c *gin.Context) {
 		primaryChannelInfo, err := d.Database.PrimaryChannelCounterparty(chainName, nextChain)
 
 		if err != nil {
-			e := deps.NewError(
-				"denom/verify-trace",
-				fmt.Errorf("failed to get primary channel for %s", hash),
-				http.StatusBadRequest,
-			)
+			if errors.Is(err, sql.ErrNoRows) {
+				cause := fmt.Sprintf("%s doesnt have primary channel for %s", chainName, nextChain)
+
+				d.LogError(
+					"channel not found",
+					"hash",
+					hash,
+					"path",
+					res.VerifiedTrace.Path,
+					"channel",
+					channel,
+					"chain",
+					chainName,
+					"err",
+					cause,
+				)
 
-			d.WriteError(c, e,
-				"cannot query primary channel information",
-				"id",
-				e.ID,
+				res.VerifiedTrace.Verified = false
+				res.VerifiedTrace.Cause = cause
+
+				c.JSON(http.StatusOK, res)
+			} else {
+				e := deps.NewError(
+					"denom/verify-trace",
+					fmt.Errorf("failed to get primary channel for %s", hash),
+					http.StatusBadRequest,
+				)
+
+				d.WriteError(c, e,
+					"cannot query primary channel information",
+					"id",
+					e.ID,
+					"hash",
+					hash,
+					"path",
+					res.VerifiedTrace.Path,
+					"chain",
+					chainName,
+					"nextChain",
+					nextChain,
+					"err",
+					err,
+				)
+			}
+			return
+		}
+
+		if primaryChannelInfo.ChannelName != channel {
+
+			cause := fmt.Sprintf("%s : not primary channel for chain %s- expecting %s got %s", hash, chainName, primaryChannelInfo.ChannelName, channel)
```

Comment:
please break it into two lines forbetter readability```suggestion
			cause := fmt.Sprintf("%s : not primary channel for chain %s- expecting %s got %s",
				hash, chainName, primaryChannelInfo.ChannelName, channel)
```